instagram/views.py

```diff
@@ -1,3 +1,26 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from .forms import NewUserForm
+from django.contrib.auth import login
+from django.contrib import messages
+
 
 # Create your views here.
+
+def activity_feed(request):
+    if request.user.is_authenticated:
+        return render(request, template_name='instagram/activity_feed.html', context={'title': 'Home!'})
+    else:
+        return redirect('login_page')
+
+
+def register(request):
+    if request.method == "POST":
+        form = NewUserForm(request.POST)
+        if form.is_valid():
+            form.save()
+            user_name = form.cleaned_data['username']
+            messages.success(request, f"An account was successfully created for {user_name}. Log in below.")
+            return redirect('login_page')
+    else:
```

Comment:
all other request types such as Put, Patch, Delete, Get will get into this block. 
you should only support GET for this block.

Reply:
Another solution is using generic views. You will implement the post and get methods and Django will take care of the other request types.